--- plugins/experimental/icap/icap_plugin.cc.orig	2022-04-11 02:13:09.624896581 +0000
+++ plugins/experimental/icap/icap_plugin.cc	2022-04-11 02:14:16.326273867 +0000
@@ -26,6 +26,7 @@
 */
 
 #include <string>
+#include <cstring>
 #include <regex>
 
 #include <netinet/in.h>
--- plugins/experimental/redo_cache_lookup/redo_cache_lookup.cc.orig	2022-04-11 02:17:00.528432143 +0000
+++ plugins/experimental/redo_cache_lookup/redo_cache_lookup.cc	2022-04-11 02:17:18.662903464 +0000
@@ -23,6 +23,7 @@
 
 #include <iostream>
 #include <regex>
+#include <cstring>
 #include <set>
 #include <sstream>
 #include <getopt.h>
